utils/data-ingestor.js

```diff
@@ -0,0 +1,229 @@
+const { find } = require('lodash');
+const df = require('dataframe-js');
+
+const specialCalcConfigs = require('../special-calculations');
+const calculateMedianError = require('../utils/calculate-median-error');
+const interpolate = require('../utils/interpolate');
+const { executeWithData: executeFormula } = require('../utils/formula');
+
+const {
+  INFLATION_FACTOR,
+  RENAME_COLS,
+  PREV_YEAR,
+  CUR_YEAR,
+} = require('../special-calculations/data/constants');
+
+/*
+ * The DataIngester class is used to convert profile data from postgres query into a Dataframe.
+ * Additionally for aggregate profiles, the DataIngester recalculates 'special' variables, configured
+ * in 'special-calculations/[profile].js'.
+ * NOTE: Dataframes (https://gmousse.gitbooks.io/dataframe-js/#dataframe-js) are immutable, so they cannot be modified
+ * in place. That is why all functions mutating a Dataframe also return one.
+ */
+class DataIngester {
+  /*
+   * Creates a DataIngester
+   * @constructor
+   * @params{Array} data - Raw data from SQL query, an array of objects representing rows
+   * @params{string} profileType - The profile type
+   * @params{Boolean} isAggregate - True if the given data selection is comprised of multiple geoids; else false
+   * @params{Boolean} isPrevious - True if the given data selection is for the older dataset year
+   */
+  constructor(data, profileType, isAggregate = false, isPrevious = false) {
+    this.data = data;
+    this.profileType = profileType;
+    this.isAggregate = isAggregate;
+    this.isPrevious = isPrevious;
+    this.extraColumns = ['codingThreshold'];
+  }
+
+  /*
+   * Main function of the DataIngestor class, turns this.data into a Dataframe
+   * with recalcuated special variables if aggregate, and with RENAME_COLS renamed
+   * if columnPrefix argument is supplied.
+   * @param{String} [columnPrefix] - String to prepend to RENAME_COLS names in Dataframe
+   * @returns{Dataframe}
+   */
+  processRaw(columnPrefix = '') {
+    let d = this.makeBaseDataFrame();
+
+    if (this.isAggregate) {
+      d = this.recalculate(d);
+    }
+
+    if (columnPrefix) {
+      d = prefixColumns(d, columnPrefix, RENAME_COLS);
+    }
+
+    return d;
+  }
+
+  /*
+   * Turns this.data (raw SQL data) into a Dataframe, and adds columns specified by this.extraColumns
+   * @returns{Dataframe}
+   */
+  makeBaseDataFrame() {
+    let d = new df.DataFrame(this.data);
+    this.extraColumns.forEach((col) => {
+      d = d.withColumn(col);
+    });
+    return d;
+  }
```

Comment:
It happened this way mostly b/c I got around to dealing with renaming the columns after I had created the class constructor. I don't really envision this class ever being instantiated and then reused; it is mostly a class so that I could cleanly & easily access the class properties across functions, without always having to pass them down. 

upon reflection, i agree -- the column prefix string should be part of the construction b/c why not!